### PR TITLE
CVE-2017-9805

### PR DESCRIPTION
**Pull request by hmeinertrita**
filled in the CVE-2017-9805 YAML file for round 1 submission

### CODE REVIEW

File: cves/CVE-2017-9805.yml [hunk 1]
```diff
@@ -45,7 +45,11 @@ description_instructions: |
   that outsiders to Struts would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  CVE-2017-9805 is a security vulnerability that could allow the execution of
+  arbitrary code through XML deserialization. When the vulnerable module
```
Comment (mll8657) [inline cves/CVE-2017-9805.yml]: Not sure if "XML deserialization" is too technical, but if so you could change it to a more general description about input which had to be interpreted/parsed through to make classes.

File: cves/CVE-2017-9805.yml [hunk 2]
```diff
@@ -119,9 +123,13 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
-  code:
-  fix:
+  answer: |
+    There were no unit tests involved in this vulnerability. There are unit
```
Comment (mll8657) [inline cves/CVE-2017-9805.yml:127]: For my vulnerability I used the same logic to argue that unit tests were involved. There were unit tests for the module, they just lacked a security test. This is something you could mention in the mistakes section as well.

**Comment by mll8657** [comment]
+3 upvotes

File: cves/CVE-2017-9805.yml (hunk 1 above)
Comment (moiseslorap) [reply cves/CVE-2017-9805.yml]: simplify into what XML deserialization means

File: cves/CVE-2017-9805.yml [hunk 3]
```diff
@@ -45,7 +45,11 @@ description_instructions: |
   that outsiders to Struts would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  CVE-2017-9805 is a security vulnerability that could allow the execution of
+  arbitrary code through XML deserialization. When the vulnerable module
+  recieved an XML payload, it does not properly validate it. This means that it
```
Comment (moiseslorap) [inline cves/CVE-2017-9805.yml]: received*

File: cves/CVE-2017-9805.yml [hunk 4]
```diff
@@ -135,10 +143,15 @@ discovered:
     The "apache" flag can be true, false, or nil.
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer:
-  date:
-  automated:
-  apache:
+  answer: |
+    The vulnerability was discovered by someone name Man Yue Mo who works for a
```
Comment (moiseslorap) [inline cves/CVE-2017-9805.yml:147]: Writing could be less impersonal.

File: cves/CVE-2017-9805.yml [hunk 5]
```diff
@@ -147,8 +160,10 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer:
-  name:
+  answer: |
+    The vulnerability was in the XML deserialization handling in the REST API
+    plugin for struts.
+  name: rest api plugin
```
Comment (moiseslorap) [inline cves/CVE-2017-9805.yml:167]: specify path

File: cves/CVE-2017-9805.yml [hunk 6]
```diff
@@ -178,38 +193,48 @@ lessons:
...
+    applies: false
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: |
+      The input was implicitly being trusted by the deserialization code, it
+      wasn't validating the input in any way, which it would have if it was
+      distrustful of the input.
```
Comment (moiseslorap) [inline cves/CVE-2017-9805.yml]: Breakdown the sentence and make it more straightforward. It is a little confusing what you are trying to say

File: cves/CVE-2017-9805.yml [hunk 7]
```diff
@@ -221,4 +246,20 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This vulnerability was mostly a design mistake. The developer
+    did not consider that the inut could potentially be malicious and could be
```
Comment (moiseslorap) [inline cves/CVE-2017-9805.yml]: input*

**Review by moiseslorap — COMMENTED** [review]
Submitted feedback

**Pull request merged**